util/webhook/webhook.go

```diff
@@ -258,6 +250,25 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}) {
 	}
 }
 
+// getWebUrlRegex compiles a regex that will match any targetRevision referring to the same repo as the given webURL.
+// webURL is expected to be a URL from an SCM webhook payload pointing to the web page for the repo.
+func getWebUrlRegex(webURL string) (*regexp.Regexp, error) {
+	urlObj, err := url.Parse(webURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse repoURL '%s'", webURL)
+	}
+
+	regexEscapedHostname := regexp.QuoteMeta(urlObj.Hostname())
+	regexEscapedPath := regexp.QuoteMeta(urlObj.Path[1:])
```

Comment:
Just a question for my understanding: Why do we quote `Path[1:]` and not the whole `Path` here?

Reply:
Good question. I think the `[1:]` index has been there since Jesse first committed the regex 4 years ago. My assumption has been that it's stripping the leading `/` from the path.

Reply:
Ah right - I see that it's already been so in the original. Yeah, in the context of the complete regexp, it makes sense, because URLs could be in different formats, including scp-format (e.g. `git@foo:bar/baz`)